plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -689,7 +689,10 @@ private void deleteModel(String modelId) {
     }
 
     private void handleException(FunctionName functionName, String taskId, Exception e) {
-        mlStats.createCounterStatIfAbsent(functionName, REGISTER, MLActionLevelStat.ML_ACTION_FAILURE_COUNT).increment();
+        if (!(e instanceof MLLimitExceededException) && !(e instanceof MLResourceNotFoundException)) {
```

Comment:
Thanks , how about also check `IllegalArgumentException` which is generally user's input is wrong, we should not treat it as some internal failure.

Reply:
Sounds good. In that case, may be we can add same exception [here](https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/utils/MLExceptionUtils.java#L44-L54) also, what do you think?

Reply:
Sure, sounds good